app/src/test/java/UseCases/GroceryUseCaseTest.java

```diff
@@ -0,0 +1,165 @@
+package UseCases;
+
+import Entities.Implementations.*;
+import Entities.RecipeItem;
+import Entities.Tag;
+import Storages.Implementations.IngredientStorageImpl;
+import Storages.Implementations.RecipeStorageImpl;
+import Storages.Implementations.TagStorageImpl;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class GroceryUseCaseTest {
```

Comment:
This test is extremely long for a class whos purpose is primarily to let other classes to the work for them. This makes it appear that maybe the SRP is being violated. As well you are likely testing "the code you wrote is the code you wrote" and not much significant features of it. Test overall behaviors without caring about HOW it was done, just that it WAS done.